types/tx/middleware.go

```diff
@@ -21,12 +22,25 @@ type ResponseSimulateTx struct {
 	Result  *sdk.Result
 }
 
+type Response struct {
+	GasWanted    uint64
+	GasUsed      uint64
+	MsgResponses []codectypes.Any // Represents each Msg service handler's response type. Will get proto-serialized into the `Data` field in ABCI, see note #2
+	Log          string
+	Events       []abci.Event
+}
+
+type Request struct {
+	Tx      sdk.Tx
+	TxBytes []byte
+}
+
 // TxHandler defines the baseapp's CheckTx, DeliverTx and Simulate respective
 // handlers. It is designed as a middleware stack.
 type Handler interface {
-	CheckTx(ctx context.Context, tx sdk.Tx, req abci.RequestCheckTx) (abci.ResponseCheckTx, error)
-	DeliverTx(ctx context.Context, tx sdk.Tx, req abci.RequestDeliverTx) (abci.ResponseDeliverTx, error)
-	SimulateTx(ctx context.Context, tx sdk.Tx, req RequestSimulateTx) (ResponseSimulateTx, error)
+	CheckTx(ctx context.Context, req Request, checkReq abci.RequestCheckTx) (Response, error)
```

Comment:
```suggestion
	CheckTx(ctx context.Context, req Request, checkReq RequestCheckTx) (Response, ResponseCheckTx, error)
```

we should use our own types:

```go
type RequestCheckTx struct {
  Type CheckTxType
}

type ResponseCheckTx {
  priority uint64
}
```